### PR TITLE
add strict flag for assert

### PR DESCRIPTION
**Pull request by susarlanikhilesh**
Addresses [Issue](https://github.com/resonatehq/resonate/issues/112)

### CODE REVIEW

**Comment by dfarr** [comment]
Thank you so much @susarlanikhilesh 🎉 🎉 

The tests are failing because some of our tests expect a panic to occur, and if the flag is not set (eg: the code is run via a test rather than cli) I believe viper is returning the default boolean value of `false`. Is there a way to check if the flag is not set and, if not, set strict to true instead?

**Comment by susarlanikhilesh** [comment]
Yes the strict flag is false, I did set when panic is true in the test cases

**Comment by dfarr** [comment]
I would prefer for `strict=true` in cases where not explicitly set. Our philosophy here is to panic by default when an unexpected event occurs, this way we can protect ourselves from being in an invalid state.

```
viper.GetBool("strict") // if not set needs to be defaulted to true
```

An easier way to do it might be to invert the name of the flag to something like `ignore-asserts` that we can default to false which makes things a little easier to implement.

File: internal/util/util.go [hunk 1]
```diff
@@ -6,10 +6,12 @@ import (
 	"time"
 
 	"github.com/robfig/cron/v3"
+	"github.com/spf13/viper"
 )
 
 func Assert(cond bool, msg string) {
-	if !cond {
+	strict := viper.GetBool("ignore-asserts")
```
Comment (dfarr) [inline internal/util/util.go]:
```suggestion
	ignoreAsserts := viper.GetBool("ignore-asserts")
```

File: internal/util/util.go [hunk 2]
```diff
@@ -6,10 +6,12 @@ import (
...
+	if !strict && !cond {
```
Comment (dfarr) [inline internal/util/util.go]:
```suggestion
	if !ignoreAsserts && !cond {
```

**Review by dfarr — CHANGES_REQUESTED** [review]

**Review by dfarr — APPROVED** [review]

**Pull request merged**